app/src/main/java/ict/ihu/gr/loopify/ApiManager.java

```diff
@@ -152,7 +157,38 @@ public void fetchTopTrackCharts(ApiResponseListener listener) {
     }
 
 
-    private static class GetJsonTask extends AsyncTask<String, Void, String> { // Async task for the retrieval of the json info about the artist and the song
+    // function to change when we migrate to Cloudfare
+    public void fetchMP3file(String youtubeURL, ApiResponseListener listener){
+        OkHttpClient client = new OkHttpClient();
+
+        // Create JSON payload
+        String json = "{ \"url\": \"" + youtubeURL + "\" }"; // the served info format our api handles
+        RequestBody body = RequestBody.create(MediaType.parse("application/json; charset=utf-8"), json);
+
+        Request request = new Request.Builder()
+                .url("http://192.168.1.3:3000/download") // we will change this to Petros's server ip/Cloudfare directory when we migrate
```

Comment:
The IP should be read from a parameter's file and not be hardcoded

Reply:
The IP and or the API keys, will all be in a parameter file and also not visible in the GitHub they will be filtered. This will take place before the migration.

Reply:
I would also [use retrofit and not simply okHttp ](https://stackoverflow.com/questions/39183294/why-use-retrofit-when-we-have-okhttp)

Reply:
Okay perfect, I will take a look at it.